dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ResourcesService.java

```diff
@@ -219,6 +219,15 @@ Result<Object> updateResource(User loginUser,
      */
     Map<String, Object> authorizeResourceTree(User loginUser, Integer userId);
 
+    /**
+     * Get resource by given resource type and full name. It return map contain resource id, name.
+     * Useful in Python API create task which need processDefinition information.
+     *
+     * @param userName user who query resource
+     * @param fullName full name of the resource
+     */
+    Map<String, Object> queryResourcesFileInfo(String userName, String fullName);
```

Comment:
Is there any way we can directly return `Resource` as return type here? I find out we already have `Resource` class in Python API code, maybe we query in python`Resource` class https://github.com/apache/dolphinscheduler/blob/59cd86157ff8d5ab25f0b200d6a30abd95a64c03/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/resource.py#L25